operator/pkg/kube/operator/operator.go

```diff
@@ -0,0 +1,240 @@
+				opts.LabelSelector = operatorLabel + "=true"
+				opts.Watch = true
+				return kubeclientset.CoreV1().ConfigMaps(namespace).Watch(opts)
+			},
+		},
+		&corev1.ConfigMap{}, 0, cache.Indexers{},
+	)
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	controller := &Controller{
+		clientExt:	clientExt,
+		namespace:      namespace,
+		informer:       configMapInformer,
+		queue:          queue,
+	}
+
+	configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			log.Infof("Processing add: %s", key)
+			if err == nil {
+				queue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			log.Infof("Processing delete: %s", key)
+			if err == nil {
+				queue.Add(key)
+			}
+		},
+	})
+
+	return controller
+}
+
+// Run starts the controller until it receves a message over stopCh.
+func (c *Controller) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	go c.informer.Run(stopCh)
+
+	// Wait for the caches to be synced before starting workers
+	log.Info("Waiting for informer caches to sync")
+	if !cache.WaitForCacheSync(stopCh, c.informer.HasSynced) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
+		return
+	}
+	wait.Until(c.runWorker, 5*time.Second, stopCh)
+}
+
+// RunWorker executes the processing loop.
+func (c *Controller) runWorker() {
+	for c.processNextItem() {
+		// continue looping
+	}
+}
+
+// processNextItem retrieves a configmap event from the cache queue.
+func (c *Controller) processNextItem() bool {
+	configMapName, quit := c.queue.Get()
+
+	if quit {
+		return false
+	}
+	defer c.queue.Done(configMapName)
+
+	err := c.processItem(configMapName.(string))
+	if err == nil {
+		// No error, reset the ratelimit counters
+		c.queue.Forget(configMapName)
+	} else if c.queue.NumRequeues(configMapName) < maxRetries {
+		log.Errorf("Error processing %s (will retry): %v", configMapName, err)
+		c.queue.AddRateLimited(configMapName)
+	} else {
+		log.Errorf("Error processing %s (giving up): %v", configMapName, err)
+		c.queue.Forget(configMapName)
+		utilruntime.HandleError(err)
+	}
+
+	return true
+}
+
+// processItem processese a configmap event.
+func (c *Controller) processItem(configMapName string) error {
+	obj, exists, err := c.informer.GetIndexer().GetByKey(configMapName)
+	if err != nil {
+		return fmt.Errorf("error fetching object %s error: %v", configMapName, err)
+	}
+
+	// This looks a little racy...
+	if exists {
+		c.addConfigMap(obj.(*corev1.ConfigMap))
+	} else {
+		c.deleteConfigMap(configMapName)
+	}
+	return nil
+}
+
+// addConfigMap is called in the event that a configmap was added to the system.
+func (c *Controller) addConfigMap(s *corev1.ConfigMap) {
+	log.Info("Converting ConfigMap to CRD object.")
```

Comment:
this particular function is the main aspect of the ConfigMap to CRD conversion.  For whatever reason, I was not able to sort out how to get YAMLtoJSON to return an array of JSON blobs.  I tried a whole boatload of different things to arrive at this particular simple implementation.

I would *really* appreciate a thorough review of this part of the code as I am going to use it within the galley PR mentioned in the commit message.

Reply:
[`yaml.NewYAMLToJSONDecoder`](https://github.com/kubernetes/apimachinery/blob/4a9a8137c0a17bc4594f544987b3f0d48b2e3d3a/pkg/util/yaml/decoder.go#L55) will parse and decode multi-doc YAML files.